rotkehlchen/accounting/history_base_entries.py

```diff
@@ -0,0 +1,257 @@
+        taxable=LedgerActionType.AIRDROP in pot.settings.taxable_ledger_actions,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    reward_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.REWARD)
+    result[reward_key] = HistoryBaseEntrySettings(
+        taxable=True,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    return result
+
+
+def history_base_entries_iterator(
+        events_iterator: Iterator[AccountingEventMixin],
+        associated_event: HistoryBaseEntry,
+) -> Iterator[HistoryBaseEntry]:
+    """
+    Takes an iterator of accounting events and transforms it into a history base entries iterator.
+    Takes associated event as an argument to be able to log it in case of errors.
+    """
+    for event in events_iterator:
+        if not isinstance(event, HistoryBaseEntry):
+            log.error(
+                f'At accounting for event {associated_event.notes} with identifier '
+                f'{associated_event.event_identifier} we expected to take an additional '
+                f'event but found a non history base entry event',
+            )
+            return
+
+        yield event
+
+
+class HistoryBaseEntriesAccountant():
+
+    def __init__(
+            self,
+            evm_accounting_aggregators: 'EVMAccountingAggregators',
+            pot: 'AccountingPot',
+    ) -> None:
+        self.evm_accounting_aggregators = evm_accounting_aggregators
+        self.pot = pot
+        self.tx_event_settings: dict[str, HistoryBaseEntrySettings] = {}
+
+    def reset(self) -> None:
+        self.evm_accounting_aggregators.reset()
+        self.tx_event_settings = (  # Using | operator is fine since keys are unique
+            self.evm_accounting_aggregators.get_accounting_settings(self.pot) |
+            make_default_accounting_settings(self.pot)
+        )
+
+    def process(
+            self,
+            event: HistoryBaseEntry,
+            events_iterator: Iterator[AccountingEventMixin],
+    ) -> int:
+        """Process a history base entry and return number of actions consumed from the iterator"""
+        timestamp = event.get_timestamp_in_sec()
+        type_identifier = event.get_type_identifier()
+        event_settings = self.tx_event_settings.get(type_identifier, None)
+        if event_settings is None:
+            event_settings = self.tx_event_settings.get(
+                event.get_type_identifier(include_counterparty=False),
+            )
+            if (  # For swaps we have a default treatment but for the rest we bail
+                event_settings is None or
+                event_settings.accounting_treatment != TxAccountingTreatment.SWAP
```

Comment:
Looking again I believe the only way to get here is by a mapping that doesn't exist but I'm still not sure why the check `event_settings.accounting_treatment != TxAccountingTreatment.SWAP`

Reply:
Hmm I also don't get it tbh. iirc it was added as part of some hotfix that was specific to swaps, but I think that it will make sense to do it for all evm events.